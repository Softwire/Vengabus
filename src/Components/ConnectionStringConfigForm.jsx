import React, { Component } from 'react';
import { serviceBusConnection } from '../AzureWrappers/ServiceBusConnection';
import { VengaServiceBusService } from '../AzureWrappers/VengaServiceBusService';
import { css } from 'react-emotion';
import { FormGroup, FormControl, ControlLabel, Button, Panel } from 'react-bootstrap';
import { blue } from '../colourScheme';
export class ConnectionStringConfigForm extends Component {
    constructor(props) {
        super(props);

        this.state = {
            value: '',
            info: ''
        };
    }

    handleChange = (event) => {
        this.setState({ value: event.target.value });
    };

    submitConnectionStringClick = () => {
        const infoPromise = VengaServiceBusService.getConnectionStringMetaData(this.state.value);
        infoPromise
            .then((response) => {
                this.setState({
                    info: {
                        NameSpaceName: response.name,
                        Status: response.status,
                        Location: response.location,
                        Permission: response.permission
                    }
                });
            })
            .catch((error) => {
                console.log(error);
            });
    };

    getValidationState = () => {};

    render() {
        const inputStyle = css`
            color: black;
        `;

        const infoBoxStyle = css`
            color: black;
            overflow-wrap: break-word;
        `;

        const buttonStyle = css`
            color: black;
            margin: 5px;
        `;
        const headerColour = css`
            background: ${blue};
        `;
        return (
            <form>
                <FormGroup controlId="formBasicText" validationState={this.getValidationState()}>
                    <ControlLabel>ServiceBus Connection String</ControlLabel>
                    <FormControl type="text" value={this.state.value} placeholder="Enter Connection String" onChange={this.handleChange} />
                    <FormControl.Feedback />
                </FormGroup>
                <Button className={buttonStyle} onClick={this.submitConnectionStringClick}>
                    Submit
                </Button>
                <div>
                    {' '}
                    <p />
                </div>
                <Panel>
                    <Panel.Heading className={headerColour}>ServiceBus Details</Panel.Heading>
                    <Panel.Body className={infoBoxStyle}>
                        <div>{`Your Connection string: ${this.state.value || ' '}`}</div>
                        <div>{`Name: ${this.state.info.NameSpaceName || ' '}`}</div>
                        <div>{`Location: ${this.state.info.Location || ' '}`}</div>
                        <div>{`Status: ${this.state.info.Status || ' '}`}</div>
                        <div>{`Permissions: ${this.state.info.Permission || ' '}`}</div>
                    </Panel.Body>
                </Panel>
            </form>
        );
    }
}
