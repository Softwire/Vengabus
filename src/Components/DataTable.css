.HighlightOnRowHover:hover {
    background-color: yellow;
}