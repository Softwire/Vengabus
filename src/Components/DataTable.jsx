import React, { Component } from "react";
import BootstrapTable from "react-bootstrap-table-next";
/*
Returns a table created from an input object
Props:
    dataToDisplay: the object which contains the data to be displayed in the table (REQUIRED)
    colProps: contains properties regarding specific columns (format: array of objects)
        dataField: specifies the name of the property of DataToDisplay which is to be displayed in the column (REQUIRED)
        text: text which is to be displayed in the header of the column (REQUIRED)
        headerStyle: object which contains CSS styling for the header (RECOMMENDED)
    rowEvents: Object which contains functions that are called on certain events of the row (OPTIONAL). For example:
                RowEvents = {
                    event: (row, rowIndex, e) => {
                        ....
                    }
                }
                where row is an object which contains all the information which is in that row
    onRowClick: Function to call when a row is clicked. Arguments are the same as above. Note that this could theoretically
                be defined in rowEvents, but if it is defined twice then an error will be thrown.

For more info see:
https://react-bootstrap-table.github.io/react-bootstrap-table2/
*/
export class DataTable extends Component {
    render() {
        const colProps = this.props.colProps;
        const dataToDisplay = this.props.dataToDisplay;
        const tableRowStyle = this.props.tableRowStyle;
        const rowEvents = this.props.rowEvents;
        const name = this.props.name;
        let onRowClick = this.props.onRowClick;
        let finalRowEvents = undefined;

        if (dataToDisplay) {
            if (!colProps) {
                throw new Error('column property object is not defined in ' + name);
            } else if (!colProps[0].dataField) {
                throw new Error('colProps.dataField is undefined in ' + name + ', cannot determine what data to display in which column.');
            } else if (!colProps[0].text) {
                throw new Error('colProps.text is undefined in ' + name + ', cannot determine header text of columns.');
            }

            if (rowEvents) {
                if (rowEvents.onClick && onRowClick) {
                    throw new Error("the onClick event for rows is defined multiple times in " + name);
                }
                onRowClick = onRowClick || rowEvents.onClick || function () { };
                finalRowEvents = { ...(rowEvents), onClick: onRowClick };
            } else if (onRowClick) {
                finalRowEvents = {
                    onClick: onRowClick
                };
            }
        }

        return dataToDisplay ? (
            <BootstrapTable
                keyField={colProps[0].dataField} //use the left-most column as the keyfield for the table
                data={dataToDisplay}
                rowClasses={tableRowStyle}
                columns={colProps}
                rowEvents={finalRowEvents}
            />
        ) : (
                <p>No data has been retrieved yet.</p>
            );
    }
}
