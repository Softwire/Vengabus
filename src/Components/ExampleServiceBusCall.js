import React, { Component } from 'react';
import { serviceBusConnection } from '../AzureWrappers/ServiceBusConnection';

export class ExampleServiceBusCall extends Component {
    constructor(props) {
        super(props);

        this.state = { retrievedData: "I haven't done it yet." };
    }

    getData = () => {
        /* In reality, we would probably want to form sending this info to the Connection object directly. But for the sake of demonstration...*/
        serviceBusConnection.setConnectionString(this.props.serviceBusConfig.connectionString);
        serviceBusConnection.setQueueName(this.props.serviceBusConfig.queueName);

        let serviceBusService = serviceBusConnection.getServiceBusService();

        serviceBusService
            .getQueue(serviceBusConnection.activeQueueName)
            .then((queueResult) => {
                let activeSettingsText = `ConnString = ${serviceBusConnection.activeServiceBusConString}. Name = ${serviceBusConnection.activeQueueName}`;
                this.setState({
                    settingsText: activeSettingsText,
                    retrievedData: JSON.stringify(queueResult)
                });
            })
            .catch(console.log);
    };

    render() {
        return (
            <div>
                <button onClick={this.getData}>Send Request</button>
                <p> I used config:'{this.state.settingsText}' </p>
                <p> I got back:'{this.state.retrievedData}' </p>
            </div>
        );
    }
}
