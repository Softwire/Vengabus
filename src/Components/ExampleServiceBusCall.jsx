import React, { Component } from 'react';
import { serviceBusConnection } from '../AzureWrappers/ServiceBusConnection';

export class ExampleServiceBusCall extends Component {
    constructor(props) {
        super(props);
        this.state = { settingsText: 'Settings not yet finalised' };
    }

    getData = () => {
        const serviceBusService = serviceBusConnection.getServiceBusService();

        serviceBusService
            .getQueue(serviceBusConnection.activeQueueName)
            .then((queueResult) => {
                const activeSettingsText = `ConnString = ${serviceBusConnection.activeServiceBusConString}. Name = ${serviceBusConnection.activeQueueName}`;
                this.setState({ settingsText: activeSettingsText });
                this.props.onDataReceive(queueResult);
            })
            .catch(console.log);

        serviceBusService
            .listQueues()
            .then((listqueueresult) => {
                const activeSettingsText = `ConnString = ${serviceBusConnection.activeServiceBusConString}. Name = ${serviceBusConnection.activeQueueName}`;
                this.setState({ settingsText: activeSettingsText });
                this.props.onDataReceive(listqueueresult);
            })
            .catch(console.log);
    };

    render() {
        return (
            <div>
                <button onClick={this.getData}>Send Request</button>
                <p>
                    I <em>actually</em> used config:'{this.state.settingsText}'
                </p>
            </div>
        );
    }
}
