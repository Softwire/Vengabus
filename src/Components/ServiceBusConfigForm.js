import React, { Component } from 'react';

export class ServiceBusConfigForm extends Component {
    render() {
        const formCredentials = this.props.formCredentials;
        const updateFormStateOnInput = this.props.updateFormStateOnInput;

        return (
            <div className="sb-config-form">
                <form>
                    Connection String:
                    <input type="text" name="connectionString" value={formCredentials.connectionString} onChange={updateFormStateOnInput} />
                    <br />
                    Queue Name:
                    <input type="text" name="queueName" value={formCredentials.queueName} onChange={updateFormStateOnInput} />
                </form>
                <p>
                    Fields currently held in form: {formCredentials.connectionString} | {formCredentials.queueName}
                </p>
            </div>
        );
    }
}
