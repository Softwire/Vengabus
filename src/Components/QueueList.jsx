import React, { Component } from 'react';
import { DataTable } from './DataTable';
import { css } from 'react-emotion';
import { paleGreyBlue, palerBlue } from '../colourScheme';

export class QueueList extends Component {
    // QQ
    // This is just a temporary function, update once the api is working for
    // getting the list of queues and individual queues
    rowClickResponseFunction(e, row, rowIndex) {
        console.log(row);
    }

    render() {
        const queueArray = this.props.queueData;

        const colProps = [
            {
                dataField: 'number',
                text: 'Number',
                headerStyle: {
                    width: '10%', textAlign: 'center'
                }
            },
            {
                dataField: 'name',
                text: 'Name',
                headerStyle: { width: '50%', textAlign: 'center' }
            },
            {
                dataField: 'status',
                text: 'Status',
                headerStyle: { width: '40%', textAlign: 'center' }
            }
        ];

        const tableRowStyle = css`
		          :hover {
		              border: 2px solid ${palerBlue};
		              background-color: ${paleGreyBlue};
		          }
              `;

        return (
            <DataTable
                name='QueueList'
                colProps={colProps}
                dataToDisplay={queueArray}
                tableRowStyle={tableRowStyle}
                onRowClick={this.rowClickResponseFunction}
            />
        );
    }
}
