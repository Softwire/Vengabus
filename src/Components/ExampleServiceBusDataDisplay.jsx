import React, { Component } from 'react';
import { QueueList } from '../Components/QueueList';
import { css } from 'react-emotion';

export class ExampleServiceBusDataDisplay extends Component {
    render() {
        const queueDivStyle = css`
            width: 30%;
            margin: 10px;
        `
        return (
            <div className={queueDivStyle}>
                <p> I got back: '{JSON.stringify(this.props.data)}' </p>
                <QueueList queueData={this.props.data} />
            </div>
        );
    }
}
