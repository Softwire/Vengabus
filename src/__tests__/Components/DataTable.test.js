import { DataTable } from "../../Components/DataTable";
import renderer from 'react-test-renderer';
import React from 'react';
import { css } from 'react-emotion';
import Adaptor from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';

configure({ adapter: new Adaptor() });

describe('DataTable', () => {

    it('displays correctly from given props', () => {

        const data = [{ number: 1, name: 'q1', status: 'active' }, { number: 2, name: 'q2', status: 'active' }, { number: 3, name: 'q3', status: 'dead' }];

        const ColProps = [
            {
                dataField: "number",
                text: "Number",
                headerStyle: { width: "10%" }
            },
            {
                dataField: "name",
                text: "Name",
                headerStyle: { width: "50%" }
            },
            {
                dataField: "status",
                text: "Status",
                headerStyle: { width: "40%" }
            }
        ];

        const RowEvents = {
            onClick: (e, row, rowIndex) => {
                console.log(row);
            }
        };

        const tableRowStyle = css`
    	          :hover {
    	              border: 2px solid grey;
    	              background-color: blue;
    	          }
    	      `;

        let dataTable = renderer.create(
            <DataTable
                ColProps={ColProps}
                DataToDisplay={data}
                RowEvents={RowEvents}
                tableRowStyle={tableRowStyle}
            />);

        expect(dataTable.toJSON()).toMatchSnapshot();
    });

});