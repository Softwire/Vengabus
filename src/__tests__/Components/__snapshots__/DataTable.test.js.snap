// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable displays correctly from given props 1`] = `
<div
  className="react-bootstrap-table"
>
  <table
    className="table table-bordered"
    id={undefined}
  >
    <thead>
      <tr
        className={undefined}
      >
        <th
          style={
            Object {
              "width": "10%",
            }
          }
        >
          Number
        </th>
        <th
          style={
            Object {
              "width": "50%",
            }
          }
        >
          Name
        </th>
        <th
          style={
            Object {
              "width": "40%",
            }
          }
        >
          Status
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="css-gyhgxo"
        onClick={[Function]}
        style={Object {}}
      >
        <td>
          1
        </td>
        <td
          onClick={[Function]}
        >
          q1
        </td>
        <td
          onClick={[Function]}
        >
          active
        </td>
      </tr>
      <tr
        className="css-gyhgxo"
        onClick={[Function]}
        style={Object {}}
      >
        <td>
          2
        </td>
        <td
          onClick={[Function]}
        >
          q2
        </td>
        <td
          onClick={[Function]}
        >
          active
        </td>
      </tr>
      <tr
        className="css-gyhgxo"
        onClick={[Function]}
        style={Object {}}
      >
        <td>
          3
        </td>
        <td
          onClick={[Function]}
        >
          q3
        </td>
        <td
          onClick={[Function]}
        >
          dead
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
